pluto_build/sql/update_empty_coord.sql

```diff
@@ -0,0 +1,34 @@
+-- Update X/Y and Lat/Long fields that are NULL with values from Lot Centroids
+-- Make sure lot centroids fall within the polygon
+SELECT bbl, centroid, point_surface,
+        _ST_CONTAINS(geom, ST_SetSRID(centroid ,4326)) as contain
+INTO update_table
+FROM(SELECT p.bbl as bbl, p.geom as geom, ST_Centroid(p.geom) as centroid, 
+	 		ST_PointOnSurface(p.geom) as point_surface
+	 FROM pluto p
+	 WHERE p.geom IS NOT NULL AND p.xcoord IS NULL) as tmp;
+
+-- if the centroid locates in the polygon, then update five fields: x/y, long/lat and centroid  
+UPDATE pluto b
+SET xcoord = ST_X(ST_Transform(t.centroid ,2263)),
+	ycoord = ST_Y(ST_Transform(t.centroid ,2263)),
+	longitude = ST_X(t.centroid),
+	latitude = ST_Y(t.centroid),
+	centroid = ST_SetSRID(t.centroid,4326)
+FROM update_table as t
+WHERE b.xcoord is NULL and b.bbl = t.bbl and t.contain is TRUE
```

Comment:
haven't seen this syntax before and curious about the functionality of t.contain here. Is it asserting it is part of the table?

Reply:
t.contain is to check whether the calculated centroid is valid or not by using the function _ST_CONTAINS and store this field in a temporary table. if the centroid is valid, we use the centroid, if not, then use the point_on_surface as an alternative.

Reply:
ah I missed the earlier definition. Thanks for pointing it out. It all makes sense now.